src/app/features/genes/components/gene-hero/gene-hero.component.spec.ts

```diff
@@ -28,12 +29,33 @@ describe('Component: Gene Hero', () => {
     fixture = TestBed.createComponent(GeneHeroComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    element = fixture.nativeElement;
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should add nomination and TEP text if nomination exists and is either is_tep or is_adi is true', () => {
+    component.gene = geneMock1;
+    fixture.detectChanges();
+    
+    const el = element.querySelector('.gene-hero-nominated') as HTMLElement;
+    
+    const expected = 'Nominated Target, Selected for Target Enabling Resource Development';
+    expect(el.textContent).toBe(expected);
+  });
+
+  it('should only add TEP text if nominations is null and either is_tep or is_adi is true', () => {
+    component.gene = geneMock3;
+    fixture.detectChanges();
+    
+    const el = element.querySelector('.gene-hero-nominated') as HTMLElement;
+    
+    const expected = 'Selected for Target Enabling Resource Development';
+    expect(el.textContent).toBe(expected);
+  });
```

Comment:
Nice tests! You could also consider adding tests to show that 1) only "Nominated Target" is shown when is_tep and is_adi are both false, and 2) nothing is displayed when nominations is null and both is_tep and is_adi are false. Up to you though!

Reply:
My latest commit should have these missing tests.  thanks!